client/http/client.go

```diff
@@ -413,6 +421,13 @@ func (c *client) WithRespHandler(
 	return &newClient
 }
 
+// WithBackoffer sets and returns a new client with the given backoffer.
+func (c *client) WithBackoffer(bo *retry.Backoffer) Client {
```

Comment:
Do we need to use a default backoff?

Reply:
We could leave this as an option to the caller. If the caller needs one, it could initialize a backoffer by itself.

Reply:
get it. Please add a test in client_test